### PR TITLE
Update to 9.4.0, 10.3.0, 11.1.0

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
This also fixes https://github.com/conda-forge/ctng-compiler-activation-feedstock/issues/42

**Comment by jakirkham** [comment]
Ah was curious if we were going to update 9.x. With 10.x is this mainly for end-users or do we want to use it in conda-forge as well? If the latter, how does 11.x play into that?

**Comment by isuruf** [comment]
We should move to gcc 10 for now. gcc 11 is needed for the runtime libraries so that system packages can be loaded on systems with latest runtime libraries.

**Comment by jakirkham** [comment]
Ah ok, thanks! :)

For some reason was thinking we would go straight to 11. When do you think it would make sense to make that move?

**Comment by jakirkham** [comment]
Also do we have an issue somewhere tracking when we will switch to GCC 10? Know there are some folks interested in having a bit of a heads up so they can prepare

cc @raydouglass

**Comment by isuruf** [comment]
> Also do we have an issue somewhere tracking when we will switch to GCC 10? 

No

**Comment by isuruf** [comment]
@beckermr, can you review?

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+  - 10.3.0
+  - 11.1.0
 libgfortran_soname:
-  - 4
+  - 5
   - 5
   - 5
 ctng_binutils:
   - 2.36
 ctng_gcc_activation_build_num:
```
Comment (beckermr) [inline recipe/conda_build_config.yaml:12]: remove this if we are not using it?

File: recipe/build.sh [hunk 2]
```diff
@@ -11,7 +27,26 @@ FINAL_DEBUG_FFLAGS=FINAL_DEBUG_FFLAGS_${ctng_target_platform_u}
...
+
+MAJOR_VERSION="${PKG_VERSION%%.*}"
+if [[ "${MAJOR_VERSION}" != 9 && "${MAJOR_VERSION}" != 10 ]]; then
+    FINAL_CXXFLAGS="$(echo $FINAL_CXXFLAGS | sed 's/-std=c++17 //g')"
+    FINAL_DEBUG_CXXFLAGS="$(echo $FINAL_DEBUG_CXXFLAGS | sed 's/-std=c++17 //g')"
+fi
+if [[ "${MAJOR_VERSION}" != 9 ]]; then
+    FINAL_FFLAGS="$(echo $FINAL_FFLAGS | sed 's/-fopenmp //g')"
+    FINAL_DEBUG_FFLAGS="$(echo $FINAL_DEBUG_FFLAGS | sed 's/-fopenmp //g')"
+fi
```
Comment (beckermr) [inline recipe/build.sh:47]: A comment on why these flags are being excluded would be good. I suspect the change to -fopenmp is breaking for some folks, for example.

File: recipe/build.sh [hunk 3]
```diff
@@ -11,7 +27,26 @@ FINAL_DEBUG_FFLAGS=FINAL_DEBUG_FFLAGS_${ctng_target_platform_u}
...
 
-if [ -z "${!FINAL_CFLAGS}" ]; then
+FINAL_CFLAGS="${!FINAL_CFLAGS}"
+FINAL_CXXFLAGS="${!FINAL_CXXFLAGS}"
+FINAL_FFLAGS="${!FINAL_FFLAGS}"
+FINAL_DEBUG_CFLAGS="${!FINAL_DEBUG_CFLAGS}"
+FINAL_DEBUG_CXXFLAGS="${!FINAL_DEBUG_CXXFLAGS}"
+FINAL_DEBUG_FFLAGS="${!FINAL_DEBUG_FFLAGS}"
+FINAL_LDFLAGS="${!FINAL_LDFLAGS}"
+FINAL_CONDA_PYTHON_SYSCONFIGDATA_NAME="${!FINAL_CONDA_PYTHON_SYSCONFIGDATA_NAME}"
```
Comment (beckermr) [inline recipe/build.sh:37]: I tried this code locally and it is making these strings empty. Is that what was intended?

Comment (beckermr) [reply recipe/build.sh:41]: ack ignore this.

**Comment by beckermr** [comment]
AFAICT I think this is good to go. The change in the openmp flags is breaking. We may want to make an announcement about it.

**Comment by isuruf** [comment]
Since there's no breakage in gcc-9, we can make an announcement when we switch to 10.

**Pull request merged**